cpp/src/IceGrid/SessionManager.h

```diff
@@ -140,19 +137,21 @@ namespace IceGrid
                         case Connect:
                             assert(!session);
                             {
-                                session = createSession(registry, timeout);
+                                assert(registry);
+                                session = createSession(*registry, timeout);
                             }
                             break;
                         case Disconnect:
                             assert(session);
-                            destroySession(session);
+                            destroySession(*session);
```

Comment:
How do we now session is not null here (and below)?

Reply:
The previous code already has an assert for this.